### PR TITLE
Fix issues #730 and #731

### PR DESCRIPTION
**Pull request by kaiboy94**

### CODE REVIEW

**Comment by kupiakos** [comment]
Links to issues: [#730](https://github.com/cs340tabyu/cs340fall2016/issues/730), [#731](https://github.com/cs340tabyu/cs340fall2016/issues/731).

File: java/src/client/join/JoinGameController.java [hunk 1]
```diff
@@ -117,9 +117,20 @@ public void reloadGamesList() {
...
                         getJoinGameView().closeModal();
                         getJoinGameView().showModal();
+                    } else if (getSelectColorView().isModalShowing() && selectedGame != null) {
+                        for (GameInfo g : games) {
+                            if (g.getId() == selectedGame.getId()) {
+                                selectedGame = g;
+                                break;
+                            }
+                        }
+                        selectedGame.getPlayers().stream()
```
Comment (kupiakos) [inline java/src/client/join/JoinGameController.java:129]: Could probably have the above logic integrated into this, but still: _niiiiiice_.

File: java/src/shared/facades/TurnFacade.java [hunk 2]
```diff
@@ -168,7 +168,7 @@ public void calcVictoryPoints() {
         for (Player p : getModel().getPlayers()) {
             int points = p.getMonuments();
             points += getModel().getMap().getPlayerSettlements(p.getPlayerIndex()).size();
-            points += getModel().getMap().getPlayerCities(p.getPlayerIndex()).size();
+            points += (getModel().getMap().getPlayerCities(p.getPlayerIndex()).size() * 2);
```
Comment (kupiakos) [inline java/src/shared/facades/TurnFacade.java:171]: Derp - that's what I get for coding this at the last minute and freaking out. Don't really need the parentheses.

File: java/src/client/join/JoinGameController.java [hunk 3]
```diff
@@ -117,9 +117,20 @@ public void reloadGamesList() {
...
+                                .filter(p -> p.getId() != getGameManager().getPlayerInfo().getId())
+                                .forEach(p -> getSelectColorView().setColorEnabled(p.getColor(), false));
```
Comment (kupiakos) [inline java/src/client/join/JoinGameController.java:131]: Will this handle when the active colors change? For example, Player 1 joins with blue, then Player 2 joins and sees that blue is not taken, then Player 1 leaves and  rejoins with green, Player 2 should now be able to choose blue and not green. Not super important right now, but maybe a TODO should belong here.

**Review by kupiakos — APPROVED** [review]

File: java/src/client/join/JoinGameController.java (hunk 3 above)
Comment (kaiboy94) [reply java/src/client/join/JoinGameController.java:131]: You're right.  New PR incoming

**Pull request merged**